DeAceroDLabs/marketplace-frontend
# feat [DMP-756] Added article not available for employees when sku exist but price is not available for employees

## What is done

[[DMP-756] click on little carousel images](https://deacero.atlassian.net/browse/DMP-756)
- Added article not available for employees when sku exist but price is not available for employees.

## How is done

- NO_AVAILABLE variable now is for when the selected characteristics do not find a SKU, this means that article is not available.
- NO_AVAILABLE_FOR_EMPLOYEES variable is for when the SKU was found but this specific aticle is not available for employees.
- Added articleSummary variable to display summary message, `Cargando..` for when is fetching price data, `NO_AVAILABLE ` message, `NO_AVAILABLE_FOR_EMPLOYEES` message, and `User still selecting` for when user is still selecting characteristics.
- Added `priceFetching` variable with useState and useEffect to detect when loadingPrice has changed.
- Add to cart button now when loading has NO action.

## How to test

1. Run the project via `npm start`.
2. Click on first Clavo and select these characteristics, you should see the price:
![image](https://user-images.githubusercontent.com/81311646/214441509-10af8c52-c93b-4089-b0bb-f476def74230.png)

3. Click on first Clavo and select these characteristics,please wait ,you should see the `Este producto no está disponible para venta a empleados.` message:

I don´t know why but this product is now available:
![image](https://user-images.githubusercontent.com/81311646/214441735-a6f9108d-6e14-45cf-a44d-c6b113eaef9c.png)
This product is still not available for employees
![image](https://user-images.githubusercontent.com/81311646/214881702-23dd99ec-93c2-480c-b293-c14f11bc7994.png)


4. Click on first Clavo and select these characteristics, you should see the `Este producto no existe`  message:

![image](https://user-images.githubusercontent.com/81311646/214441899-5f21b035-29eb-412a-9c4d-7c58cabb688f.png)
